src/navigation/search.py

```diff
@@ -11,37 +11,69 @@
 from trajectory import Trajectory
 from mrover.msg import GPSPointList
 from util.ros_utils import get_rosparam
+from math import ceil
 
 
 @dataclass
 class SearchTrajectory(Trajectory):
     # Associated fiducial for this trajectory
     fid_id: int
-    # Helper for building spiral
-    dirs: ClassVar[np.ndarray] = np.array([[0, -1], [-1, 0], [0, 1], [1, 0]])
 
     @classmethod
-    def spiral_traj(cls, center: np.ndarray, num_turns: int, distance: int, fid_id: int) -> SearchTrajectory:
+    def get_polar_spiral_formula(
+        cls, distance_between_spirals: float
+    ) -> Callable[[Union[np.ndarray, float]], Union[np.ndarray, float]]:
+        """
+        Returns a polar function that takes in an angle and returns the radius of a spiral
+        such that the distance between each spiral is 'distance_between_spirals'
+        :param distance_between_spirals:    distance between each spiral (float)
+        :return:    polar function (callable that takes in either a float angle or an array of angles and outputs the radius at that angle or angles)
+        """
+        return lambda theta: theta * (distance_between_spirals / (2 * np.pi))
+
+    @classmethod
+    def gen_spiral_coordinates(
+        cls, coverage_radius: float, distance_between_spirals: float, num_points_per_spiral: int
+    ) -> np.ndarray:
+        """
+        Generates a set of coordinates for a spiral search pattern centered at the origin
+        :param coverage_radius:     radius of the spiral search pattern (float)
+        :param distance_between_spirals:    distance between each spiral (float)
+        :param num_points_per_spiral:   number of points per spiral (int)
+        :return:    np.ndarray of coordinates
+        """
+        num_spirals = ceil(coverage_radius / distance_between_spirals) + 1
+        angles = np.linspace(0, 2 * np.pi * num_spirals, num_points_per_spiral * num_spirals + 1)
+        radii = cls.get_polar_spiral_formula(distance_between_spirals)(angles)
```

Comment:
going off what I said earlier, you could just replace this line with the contents of the lambda function, and it would do the same thing while being much less confusing imo